aggregator/leader_flush_mgr.go

```diff
@@ -257,6 +274,29 @@ func (mgr *leaderFlushManager) persistFlushTimes(flushTimes *schema.ShardSetFlus
 	mgr.metrics.flushTimesPersist.ReportSuccessOrError(persistErr, mgr.nowFn().Sub(persistStart))
 }
 
+func (mgr *leaderFlushManager) ownedShards() (shard.Shards, error) {
+	stagedPlacement, onStagedPlacementDoneFn, err := mgr.placementWatcher.ActiveStagedPlacement()
+	if err != nil {
+		mgr.metrics.activeStagedPlacementErrors.Inc(1)
+		return nil, err
+	}
+	defer onStagedPlacementDoneFn()
```

Comment:
Do we still want to call the onStagedPlacementDone function if line 285 fails? Does it need to do anything different if further actions fail?

Reply:
Yes `onStagedPlacementDone` needs to be called as long as the `.ActiveStagedPlacement()` call succeeds, regardless of whether further actions fail or not.